adapter/src/main/java/com/twilio/kudu/sql/metadata/KuduRelMdSelectivity.java

```diff
@@ -53,26 +53,31 @@ public Double getSelectivity(RelNode rel, RelMetadataQuery mq, @Nullable RexNode
     final RexExecutor executor = Util.first(rel.getCluster().getPlanner().getExecutor(), RexUtil.EXECUTOR);
     final RexSimplify rexSimplify = new RexSimplify(rel.getCluster().getRexBuilder(), RelOptPredicateList.EMPTY,
         executor);
-    RexNode e = rexSimplify.simplify(predicate);
-    if (e.getKind() == SqlKind.SEARCH) {
-      RexCall call = (RexCall) e;
-      Sarg sarg = ((RexLiteral) call.getOperands().get(1)).getValueAs(Sarg.class);
-      double selectivity = 0.d;
-      for (Object o : sarg.rangeSet.asRanges()) {
-        Range r = (Range) o;
-        if (!(r.lowerEndpoint() instanceof TimestampString) || !(r.upperEndpoint() instanceof TimestampString)) {
-          return RelMdUtil.guessSelectivity(predicate);
-        }
-        long lowerBound = r.hasLowerBound() ? ((TimestampString) r.lowerEndpoint()).getMillisSinceEpoch() : 0l;
-        long upperBound = r.hasUpperBound() ? ((TimestampString) r.upperEndpoint()).getMillisSinceEpoch()
-            : System.currentTimeMillis();
-        selectivity += (upperBound - lowerBound) / MAX_TIME_RANGE;
-        if (selectivity > 1.0) {
-          return 1.0;
+    RexNode simplifiedPred = rexSimplify.simplify(predicate);
+    double sel = 1.0;
+    for (RexNode pred : RelOptUtil.conjunctions(simplifiedPred)) {
```

Comment:
Does this change calculate the rangeSelectivity based on all fields in where clause instead of just 1?

Reply:
Previously if the selectivity would be modified only if the WHERE clause had a single range filter, this changes it so that if you have an AND in the where clause it would get matched and the selectivity would be modified.